### PR TITLE
Edit schedule to master

### PR DESCRIPTION
**Pull request by bfok123**
Starting my pull request for edit schedule and updated schedule component to master.
I have mostly finished the edit schedule screen and updated the schedule component to work with group schedules as well (different shades of color depending on how many people have overlapping timeslots) by providing a boolean prop to it (isGroupSchedule). I also connected the edit schedule screen with the FAB on the Profile screen.

The majority of files changed are because I deleted the node_modules folder that was accidentally added to the root folder of our project. You can ignore those.

Note that the schedule component takes in a schedule in the format:
[ { startTime: Date, endTime: Date }, ... ]

Todos:
- Use time picker instead of react-native-paper menus for selecting start and end times
- Add backend and frontend controllers for controlling user schedules

### CODE REVIEW

File: frontend/components/reference_component/index.js [hunk 1]
```diff
@@ -7,12 +7,17 @@ import { View } from 'react-native';
 import styles from './styles';
 import { Button, Text } from 'react-native-paper';
```
Comment (apratan) [inline frontend/components/reference_component/index.js:8]: We should just remove this class. It's no longer necessary

Comment (bfok123) [reply frontend/components/reference_component/index.js]: Removed reference controller, component, and screen.

File: api/server/controllers/userController.js [hunk 2]
```diff
@@ -304,3 +311,48 @@ exports.removeMeeting = function(req, res) {
...
+        data: data
+      });
+    }
+  });
+}
+
+// Receives userId and schedule as input
+// Sets the schedule for the user to the given schedule
+exports.setSchedule = function(req, res) {
+  User.update(
```
[the remark this answers is not in the document]
Comment (bfok123) [reply api/server/controllers/userController.js]: Changed to updateOne.

File: api/server/controllers/userController.js [hunk 3]
```diff
@@ -149,6 +151,10 @@ exports.addGroupRequest = function(req, res) {
         message: "Request successfully added to the user",
         data: data
       })
+      // console.log(io);
+      console.log(socket.clients);
```
[the remark this answers is not in the document]
Comment (bfok123) [reply api/server/controllers/userController.js]: Removed testing code.

**Pull request merged**